sdk/core/core-http/src/httpHeaders.ts

```diff
@@ -28,10 +28,42 @@ export interface HttpHeader {
  */
 export type RawHttpHeaders = { [headerName: string]: string };
 
+export interface HttpHeadersLike {
```

Comment:
Do we need to document these fields?

Reply:
Yeah I should copy the docs from the implementation! I will update with this